src/RestServer/RestServerUtility.JTokens.cs

```diff
@@ -0,0 +1,330 @@
+// modifications are permitted.
+
+using Neo.Network.P2P.Payloads;
+using Neo.Network.P2P.Payloads.Conditions;
+using Neo.SmartContract;
+using Neo.SmartContract.Manifest;
+using Newtonsoft.Json.Linq;
+
+namespace Neo.Plugins.RestServer
+{
+    public static partial class RestServerUtility
+    {
+        public static JToken BlockHeaderToJToken(Header header, global::Newtonsoft.Json.JsonSerializer serializer) =>
+            JToken.FromObject(new
+            {
+                header.Timestamp,
+                header.Version,
+                header.PrimaryIndex,
+                header.Index,
+                header.Nonce,
+                header.Hash,
+                header.MerkleRoot,
+                header.PrevHash,
+                header.NextConsensus,
+                Witness = WitnessToJToken(header.Witness, serializer),
+                header.Size,
+            }, serializer);
+
+        public static JToken WitnessToJToken(Witness witness, global::Newtonsoft.Json.JsonSerializer serializer) =>
+            JToken.FromObject(new
+            {
+                witness.InvocationScript,
+                witness.VerificationScript,
+                witness.ScriptHash,
+            }, serializer);
+
+        public static JToken BlockToJToken(Block block, global::Newtonsoft.Json.JsonSerializer serializer) =>
+            JToken.FromObject(new
+            {
+                block.Timestamp,
+                block.Version,
+                block.PrimaryIndex,
+                block.Index,
+                block.Nonce,
+                block.Hash,
+                block.MerkleRoot,
+                block.PrevHash,
+                block.NextConsensus,
+                Witness = WitnessToJToken(block.Witness, serializer),
+                block.Size,
+                Transactions = block.Transactions.Select(s => TransactionToJToken(s, serializer)),
+            }, serializer);
+
+        public static JToken TransactionToJToken(Transaction tx, global::Newtonsoft.Json.JsonSerializer serializer) =>
+            JToken.FromObject(new
+            {
+                tx.Hash,
+                tx.Sender,
+                tx.Script,
+                tx.FeePerByte,
+                tx.NetworkFee,
+                tx.SystemFee,
+                tx.Size,
+                tx.Nonce,
+                tx.Version,
+                tx.ValidUntilBlock,
+                Witnesses = tx.Witnesses.Select(s => WitnessToJToken(s, serializer)),
+                Signers = tx.Signers.Select(s => SignerToJToken(s, serializer)),
+                Attributes = tx.Attributes.Select(s => TransactionAttributeToJToken(s, serializer)),
+            }, serializer);
+
+        public static JToken SignerToJToken(Signer signer, global::Newtonsoft.Json.JsonSerializer serializer) =>
+            JToken.FromObject(new
+            {
+                Rules = signer.Rules.Select(s => WitnessRuleToJToken(s, serializer)),
```

Comment:
```suggestion
                Rules = signer.Rules?.Select(s => WitnessRuleToJToken(s, serializer)),
```
In `WalletTransferAssets` for normal NEO or GAS transactions, no `Rules` here.

Reply:
can you do
```
Rules = signer.Rules != null signer.Rules.Select(s => WitnessRuleToJToken(s, serializer)) : Array.Empty<WitnessRule>(),
```

So that way, it returns an empty array on null. So its consistent with the rest of the json in `RestServer`.